libminifi/include/EventDrivenSchedulingAgent.h

```diff
@@ -39,21 +41,28 @@ class EventDrivenSchedulingAgent : public ThreadedSchedulingAgent {
    * Create a new event driven scheduling agent.
    */
   EventDrivenSchedulingAgent(std::shared_ptr<core::controller::ControllerServiceProvider> controller_service_provider, std::shared_ptr<core::Repository> repo,
-                             std::shared_ptr<core::Repository> flow_repo, std::shared_ptr<core::ContentRepository> content_repo, std::shared_ptr<Configure> configuration)
-      : ThreadedSchedulingAgent(controller_service_provider, repo, flow_repo, content_repo, configuration) {
-  }
-  // Destructor
-  virtual ~EventDrivenSchedulingAgent() {
+                             std::shared_ptr<core::Repository> flow_repo, std::shared_ptr<core::ContentRepository> content_repo, std::shared_ptr<Configure> configuration,
+                             std::shared_ptr<utils::ThreadPool<utils::ComplexTaskResult>> thread_pool)
+      : ThreadedSchedulingAgent(controller_service_provider, repo, flow_repo, content_repo, configuration, thread_pool) {
+    int slice = configuration->getInt(Configure::nifi_flow_engine_event_driven_time_slice, DEFAULT_TIME_SLICE_MS);
+    if (slice < 10 || 1000 < slice) {
+      throw Exception(FLOW_EXCEPTION, std::string(Configure::nifi_flow_engine_event_driven_time_slice) + " is out of reasonable range!");
```

Comment:
0 doesn't seem to be valid for me as the processor could only process 1 FF before getting back to the end of the queue, that looks pretty bad for me. 
On the other end, lasting more than 1s seems like starvation for me, I don't think that should happen. In case that happens, I would consider that as an error.